src/wrappers/themis/android/build.gradle

```diff
@@ -0,0 +1,108 @@
+apply plugin: 'com.android.library'
+
+buildscript {
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.2.1'
+
+        // Two necessary plugins for uploading .aar to bintray
+        // from: https://android.jlelse.eu/how-to-distribute-android-library-in-a-convenient-way-d43fb68304a7
+        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.4'
+        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.4.1'
+    }
+}
+
+dependencies {
+    implementation project(":boringssl")
+    // Instrumentation tests
+    androidTestImplementation 'androidx.test:runner:1.1.0'
+    androidTestImplementation 'androidx.test:rules:1.1.0'
+}
+
+android {
+    compileSdkVersion 28
+    buildToolsVersion "28.0.3"
+
+    // BoringSSL requires at least API 21. Google Play as of August 2019 requires
+    // to target at least API 28 (but we can still support lower versions).
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 28
+        // Our tests are written in JUnit, set default runner appropriately
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    sourceSets {
+        main {
+            java.srcDirs = ['../java']
+            manifest.srcFile 'AndroidManifest.xml'
+            jniLibs.srcDirs = ['libs']
+        }
+        androidTest.setRoot('tests/themis/wrappers/android')
+        androidTest.java.srcDirs = ['../../../../tests/themis/wrappers/android']
+        androidTest.manifest.srcFile '../../../../tests/themis/wrappers/android/AndroidManifest.xml'
+    }
+
+    // Dependencies for instrumentation tests
+    useLibrary 'android.test.runner'
+    useLibrary 'android.test.base'
+    useLibrary 'android.test.mock'
+
+    // ensure we execute boringssl tasks first
+    // tasks.whenTaskAdded({Task task -> task.dependsOn('boringssl:' + task.name)})
+
+    // publishing and bitray upload tasks should not run for ':boringssl' project
+    tasks.whenTaskAdded { task ->
+        println "executing $task ..."
+        if (task.name != 'bintrayUpload' && task.name != 'publishProductionPublicationToMavenLocal' && task.name != 'generatePomFileForProductionPublication') {
+            task.dependsOn(':boringssl:' + task.name)
+        }
+    }
+
+    externalNativeBuild {
+        ndkBuild {
+            path "../../../../jni/Android.mk"
+        }
+    }
+}
+
+// distribution
+
+apply plugin: 'com.jfrog.bintray'
+apply plugin: 'maven-publish'
+
+publishing {
+    publications {
+        Production(MavenPublication) {
+            artifact("$buildDir/outputs/aar/projects-release.aar")
+            groupId 'com.cossacklabs.com'
+            artifactId 'themis'
+            version '0.12.0'
```

Comment:
I had a dream that we moved versions to a single source / constant...

Reply:
Yeah, I thought about that because updating it in _three places_ is too close to my personal DRY rule of extracting common code when there are at least three uses.

I guess it should be possible to move the version to Gradle properties (since we have the file now) and set it from there for all subprojects.

Reply:
Yes, I had same thoughts. It’s not critical, but maybe if on some point we decide to have one version, it would be nice. 

I was thinking if `java-themis` and `android-themis` can potentially have different versions, say 0.13.1 and 0.13.2? Due to different workarounds/hotfixes in each.

Reply:
> if java-themis and android-themis can potentially have different versions, say 0.13.1 and 0.13.2?

Valid point. Typically, if we need to change the code then both will get an update, but there's also Android build system to consider. For example, if Android adds a new device architecture then it's a good idea to ship a patch update with the same code built for that new architecture, which only makes sense for Android.

Anyhow, I think it is beneficial to have the versions in one place just so that we don't have to run around the repository during regular releases. We can, for example, have two properties: `javaThemisVersion` and `androidThemisVersion`.

The properties seem to work (fdc0d23a4f28373c6cac55a72e93f19b6facdeea), I'm currently testing if they work for Android too, just in case.

Reply:
Or maybe keep the versions in individual build.gradle files, but extract them into a constant somewhere close to the top so that it's visible and quick to change? IMO, it makes equal sense to keep the version in the “project file”.

At least we won't be duplicating the string in the Android file.

Reply:
Having property will increase our UX as maintainers significantly. When we will need different values, we’ll introduces two properties.

Also, pls don’t forget to update internal release guidelines section where we enumerate versions :D

Reply:
I'll add two properties from the get go then.

Maybe in some parallel dimension the [VERSION](https://github.com/cossacklabs/themis/blob/master/VERSION) file looks like this:
```
Core:       0.13.0
GoThemis:   0.13.0
PyThemis:   0.13.1
JavaThemis: 0.13.2
```
And you don’t have to `git grep ${current_minor_version}` to see if anything is missing an update 💤 ☁️ 🌈 🦄 

> pls don’t forget to update internal release guidelines section where we enumerate versions

Sure, will do.